python/lsst/ap/association/dia_collection.py

```diff
@@ -19,3 +19,188 @@
+        """ Update the internal searchable spatial tree on the DIAObjects.
+
+        Returns
+        -------
+        None
+        """
+        self._is_valid_tree = False
+
+        # Create searchable spatial tree.
+
+        self._is_valid_tree = True
+
+        return None
+
+    def append(self, dia_object):
+        """ Add a new DIAObject to this collection.
+
+        Parameters
+        ----------
+        dia_object : A DIAObject class instance
+            Input dia_object to append to this collection.
+
+        Returns
+        -------
+        None
+        """
+
+        self._is_updated = False
+        self._is_valid_tree = False
+
+        self.dia_objects.append(dia_object)
+
+        return None
+
+    def score(self, dia_source_catalog, max_dist_arcsec):
+        """ Compute a quality score for each dia_source/dia_object pair
+        between this collection and an input diat_source catalog.
+
+        max_dist sets maximum seperation in arcseconds to consider a
+        dia_source a possible match to a dia_object. If the pair is
+        beyond this distance no score is computed.
+
+        Parameters
+        ----------
+        dia_source_catalog : an lsst.afw.SourceCatalog
+            A contiguous catalo of dia_sources to "score" based on distance
+            and (in the future) other metrics.
+        max_dist_arcsec : float
+            Maximum allowed distance to compute a score for a given DIAObject
+            DIASource pair.
+
+        Returns
+        -------
+        (N DIAObject by M DIASource) float array of between DIAObjects and
+        DIASources.
+        """
+        pass
+
+    def match(self, dia_source_catalog, scores):
+        """ Append DIAsources to DIAObjectds given a score and create new
+        DIAObjects in this collection from DIASources with poor scores.
+
+        Parameters
+        ----------
+        dia_source_catalog : an lsst.afw.SourceCatalog
+            A contiguous catalo of dia_sources for which the set of scores
+            has been computed on with DIAObjectCollection.score.
+        scores : (N DIAObject by M DIASource) float array
+            A ndarray containing the scores between each DIAObject and
+            each DIASource.
+
+        Returns
+        -------
+        Indices of newly updated and created DIAObjects
+        """
+        pass
+
+    @property
+    def is_updated(self):
+        """ Return the status of the internal DIAObjects and if their summary
+        statistics have been properly updated.
+
+        Return
+        ------
+        bool
+        """
+
+        return self._is_updated
+
+    @property
+    def is_valid_tree(self):
+        """ Return the status of the internal spatial search tree.
+
+        If the tree has not been updated with the current positions of
+        all DIAObjects internal to this collection we return false.
+
+        Return
+        ------
+        bool
+        """
+
+        return self._is_valid_tree
```

Comment:
Sure. I think this is something that gets taken care of by the future AssociationTask. On instantiation this would query where ever the DIAObjects are stored and then create the DIAObjectCollection for the association step. AssociationTask would then take care of writing the data as well.